app/views/assignments/show.html.erb

```diff
@@ -32,206 +32,201 @@
-                <%= link_to t('groups.members.decline'),
-                            decline_invitation_course_assignment_groups_path(course_id: @current_course.id,
-                                                                             assignment_id: @assignment.id,
-                                                                             grouping_id: grouping.id),
-                            { class: 'button',
-                              method: :patch, data: { confirm: t('groups.members.decline_invitation_confirm') } } %>
-              <% end %>
-            </div>
+                <ul>
+                  <% grouping.student_memberships.includes(role: :user).each do |member|%>
+                    <%= render partial: 'assignments/member',
+                               locals: { member: member,
+                                         grouping: grouping } %>
+                  <% end -%>
+                </ul>
+                <% if !@assignment.scanned_exam && @assignment.past_collection_date?(@current_role.section) %>
+                  <p class='information'>
+                    <%= t('groups.student_interface.may_not_join') %>
+                  </p>
+                <% else %>
+                  <%= link_to t('groups.members.join'),
+                              accept_invitation_course_assignment_groups_path(course_id: @current_course.id,
+                                                                              assignment_id: @assignment.id,
+                                                                              grouping_id: grouping.id),
+                              { method: :patch, class: 'button' } %>
+                  <%= link_to t('groups.members.decline'),
+                              decline_invitation_course_assignment_groups_path(course_id: @current_course.id,
+                                                                               assignment_id: @assignment.id,
+                                                                               grouping_id: grouping.id),
+                              { class: 'button',
+                                method: :patch, data: { confirm: t('groups.members.decline_invitation_confirm') } } %>
+                <% end %>
+              </div>
+            <% end %>
           <% end %>
-        <% end %>
 
-        <% # If students are allowed to form groups and
-           # the grace period for this assignment has not yet passed,
-           # render UI elements for group creation
-           if !@assignment.past_collection_date?(@current_role.section) && @assignment.student_form_groups %>
+          <% # If students are allowed to form groups and
+             # the grace period for this assignment has not yet passed,
+             # render UI elements for group creation
+             if !@assignment.past_collection_date?(@current_role.section) && @assignment.student_form_groups &&
+               @assignment.group_max > 1 %>
```

Comment:
This change isn't necessary since you added the check for the `group_max > 1` at the top level.